modules/programs/neovim.nix

```diff
@@ -24,12 +24,36 @@ let
     merge = mergeOneOption;
   };
 
+  pluginWithConfigType = types.submodule {
+    options = {
+      plugin = mkOption {
+        type = types.package;
+        description = "vim plugin";
+      };
+      config = mkOption {
+        type = types.lines;
+        description = "vimscript for this plugin to be placed in init.vim";
+        default = "";
+      };
+    };
+  };
+
+  # A function to get the configuration string (if any) from an element of 'plugins'
+  pluginConfig = p:
+    if builtins.hasAttr "plugin" p && builtins.hasAttr "config" p then ''
+      """"""""""""""""""""""""""""""""
+      " ${p.plugin.pname}
```

Comment:
what I like to do is my init.vim is `set foldmethod=marker` and then use default markers like
```
" ${p.plugin.pname} {{{
${p.config}
" }}}
```
what do you think ? It's frowned upon to pass a function as an option and making it configurable otherwise may be a headache so overall, it may be easier to just put p.config and assume the user prepared it as he wished:
`fugitive.config = prepareConfig "let g:fugitive_XX=1"`

Reply:
I don't really mind what the separation is, but I don't really want to have to litter my config with heaps of `prepareConfig` or do a little dance mapping over the attribute sets. TBH I very rarely inspect the generated init.vim code anyway.

Reply:
I do look at it though (and like the ability to fold/unfold). The most generic solution is to just dump ${p.config} but if you don't like it 1/ you can do as you wish and this can be revisited later. Changes would be backwards compatible anyway 2/ we wait for a 3rd opinion :)

Reply:
@teto, sorry took a while. I've corrected things with your suggestion, thanks!